### PR TITLE
attempted on how transparant images work

### PR DESCRIPTION
**Pull request by nielsnl68**
# What does this implement/fix?

This PR will change the way how transparant images are stored and shown using the image method.

It will use one color code to define which color is seen as transparant. 

**Before this can be merge i wish to have some testers and ideas on how to make it simpler.**

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/display/display_buffer.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@ static const char *const TAG = "display";
 
 const Color COLOR_OFF(0, 0, 0, 0);
 const Color COLOR_ON(255, 255, 255, 255);
+const Color COLOR_TRANSPARANT(0, 0, 0, 255);
```
Comment (guillempages) [inline esphome/components/display/display_buffer.cpp]:
```suggestion
const Color COLOR_TRANSPARENT(0, 0, 0, 255);
```

Comment (nielsnl68) [reply esphome/components/display/display_buffer.cpp]: thanks

**Comment by guillempages** [comment]
You might want to take a look at https://github.com/esphome/issues/issues/4226; it looks like the change you did with the TRANSPARENT_BINARY broke something on e-ink displays

**Comment by nielsnl68** [comment]
> You might want to take a look at [esphome/issues#4226](https://github.com/esphome/issues/issues/4226); it looks like the change you did with the TRANSPARENT_BINARY broke something on e-ink displays

Could be, that is why i turned this one into a draft version and that i was eager to ask your input from you as well.

File: esphome/components/image/__init__.py [hunk 2]
```diff
@@ -69,44 +69,63 @@ async def to_code(config):
...
             data[pos] = pix[2]
             pos += 1
 
-    elif config[CONF_TYPE] == "RGB565":
-        image = image.convert("RGB")
+    elif config[CONF_TYPE] in ["RGB565", "TRANSPARENT_IMAGE"]:
+        image = image.convert("RGBA")
         pixels = list(image.getdata())
-        data = [0 for _ in range(height * width * 3)]
+        data = [0 for _ in range(height * width * 1)]
```
Comment (nielsnl68) [inline esphome/components/image/__init__.py]:
```suggestion
        data = [0 for _ in range(height * width * 2)]
```

**Comment by nielsnl68** [comment]
This PR is now integrated into a new pr (#4600)

**Pull request closed without merging**